lib/new_relic/telemetry/redix.ex

```diff
@@ -0,0 +1,185 @@
+defmodule NewRelic.Telemetry.Redix do
+  use GenServer
+
+  def start_link() do
+    enabled = NewRelic.Config.feature?(:redix_instrumentation)
+    GenServer.start_link(__MODULE__, [enabled: enabled], name: __MODULE__)
+  end
+
+  @redix_connection [:redix, :connection]
+  @redix_pipeline_stop [:redix, :pipeline, :stop]
+
+  @init_events [@redix_connection]
+  @connected_events [@redix_connection, @redix_pipeline_stop]
+
+  def init(enabled: false), do: :ignore
+
+  def init(enabled: true) do
+    config = %{
+      handler_id: {:new_relic, :redix},
+      connections: %{},
+      collect_db_query?: NewRelic.Config.feature?(:db_query_collection)
+    }
+
+    :telemetry.attach_many(
+      config.handler_id,
+      @init_events,
+      &__MODULE__.handle_event/4,
+      config
+    )
+
+    Process.flag(:trap_exit, true)
+    {:ok, config}
+  end
+
+  def handle_call({@redix_connection, meta}, _from, config) do
+    [host, port] = meta.address |> String.split(":")
+
+    config =
+      update_in(
+        config.connections,
+        &Map.put(
+          &1,
+          meta.connection,
+          %{address: meta.address, host: host, port: port}
+        )
+      )
+
+    :telemetry.detach(config.handler_id)
```

Comment:
Attach with additional handlers instead of re-attaching. Each handler will contain just it's config.